assets/src/application/ui/input/dateTime/withBaseProvider.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+import BaseUIProvider from './BaseUIProvider';
+
+const withBaseProvider = <Props extends {}>(Component: React.ComponentType<Props>): React.FC<Props> => (
+	props: Props
+) => {
+	return (
+		<BaseUIProvider>
+			<Component {...props} />
+		</BaseUIProvider>
+	);
+};
+
+export default withBaseProvider;
```

Comment:
imho this could be part of the infrastructure adapter for the datepicker and timepicker and wrap the exports for those adapters right at the source